docs/topics/settings.rst

```diff
@@ -604,6 +604,26 @@ If you want to disable it set to 0.
 
     This feature needs Twisted >= 11.1.
 
+.. setting:: DOWNLOAD_ALLOW_DATALOSS
+
+DOWNLOAD_ALLOW_DATALOSS
+-----------------------
+
+Default: ``False``
+
+Whether or not to allow broken responses, where declared ``Content-Length``
+does not match content sent by the server. If ``False``, these responses
+trigger a ``ResponseFailed`` error. If ``True``, these responses are allowed
```

Comment:
> Would it be useful to have an option to retry these responses, but keep the last one if retry attempts failed? 

Seems like this feature needs to be discussed on its own issue.

> can retries fix this data loss problem, or is it usually just a server configuration issue?

There are several scenarios where this can happe. For example, if the server returns a static file, sets the ``Content-Length`` correctly but fails to read a part of the file due to data corruption or disk failure. The common misconfiguration case seems to be when the application uses a content compression filter which is enabled after the ``Content-Length`` is set.

We can't tell how often a retry fixed this issue unless we mine past logs.